cli/lsp/analysis.rs

```diff
@@ -352,21 +351,14 @@ fn is_preferred(
 
 /// Convert changes returned from a TypeScript quick fix action into edits
 /// for an LSP CodeAction.
-async fn ts_changes_to_edit<F, Fut, V>(
+async fn ts_changes_to_edit(
   changes: &[tsc::FileTextChanges],
-  index_provider: &F,
-  version_provider: &V,
-) -> Result<Option<lsp::WorkspaceEdit>, AnyError>
-where
-  F: Fn(ModuleSpecifier) -> Fut + Clone,
-  Fut: Future<Output = Result<LineIndex, AnyError>>,
-  V: Fn(ModuleSpecifier) -> Option<i32>,
-{
+  language_server: &mut language_server::Inner,
```

Comment:
I opted to remove the callbacks (because they were cumbersome to change) and call back directly - and change `pub` to `pub(crate)` because `Inner` is `pub(crate)`. It's kinda closely coupled but I feel that's okay.

I considered breaking it out into a trait but that felt like premature optimization as long as there's a single consumer.

Reply:
Yeah, that makes sense... I originally did it because I couldn't easily pass in the language server because of the multiple versions of the arc/mutex's and the multi threading versus any real desire for abstraction.